pennylane/transforms/tape_expand.py

```diff
@@ -25,6 +27,11 @@
     not_tape,
 )
 
+# Needed for custom decomposition context manager
+from pennylane.transforms.qfunc_transforms import NonQueuingTape
+
+NonQueuingTape = type("NonQueuingTape", (NonQueuingTape, qml.tape.QuantumTape), {})
```

Comment:
Curious what that means?

Reply:
Ah, currently, the `NonQueueingTape` is a mixin, so here we use `type(name, bases, dict)` to 'generate' a class that inherits from both `NonQueueingTape` and `QuantumTape`.


This is because currently there are _multiple_ tape classes, and this allows us to make any of the tape subclasses non-queuing :(

However, once these tape subclasses are deleted, this simplifies things significantly, and we can remove the need to do this and simply allow `NonQueuingTape` to be imported and used directly.